example/index.js

```diff
@@ -32,6 +32,21 @@ const start = async () => {
     }
   })
 
+  fastify.route({
+    url: '/with-hooks',
+    method: 'GET',
+    handler: async function routeHandler (request, reply) {
+      return 'Hello World With hooks!'
+    },
+    preValidation: function preValidation (request, reply, done) {
+      const { activeSpan } = request.openTelemetry()
+
+      activeSpan.addEvent('preValidation')
```

Comment:
Could we have this example start and end a new span instead? Would like to show that spans from hooks are now automatically children of the plugin's span 😎